src/Functions/FunctionsHashing.h

```diff
@@ -465,6 +465,26 @@ struct GccMurmurHashImpl
     static constexpr bool use_int_hash_for_pods = false;
 };
 
+/// To be compatible with Kafka: https://github.com/apache/kafka/blob/461c5cfe056db0951d9b74f5adc45973670404d7/clients/src/main/java/org/apache/kafka/common/utils/Utils.java#L480
+struct KafkaMurmurHashImpl
+{
+    static constexpr auto name = "kafkaMurmurHash";
+
+    using ReturnType = Int32;
+
+    static Int32 apply(const char * data, const size_t size)
+    {
+        return static_cast<ReturnType>(MurmurHash2(data, size, 0x9747b28cU));
```

Comment:
The kafka murmurhash always returns a positive numerical value, I would suggest something like this

`    
    using ReturnType = UInt32;

    static UInt32 apply(const char * data, const size_t size)
    {
        return MurmurHash2(data, size, 0x9747b28cUL) & 0x7fffffff;
    }
`

Reply:
Are the unit tests to Murmur in the Kafka repository hash wrong then?https://github.com/apache/kafka/blob/139f7709bd3f5926901a21e55043388728ccca78/clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java#L93
Or how is it possible to check the results of this function from inside Kafka?

Reply:
I specifically changed the return type to correspond with Kafka's version that returns Int32.

Reply:
Yep, though original murmur2 function returns Int32, it is automatically casted to UInt32 in DefaultPartitioner: https://github.com/apache/kafka/blob/139f7709bd3f5926901a21e55043388728ccca78/clients/src/main/java/org/apache/kafka/clients/producer/internals/BuiltInPartitioner.java#L328